web/packages/teleterm/src/services/tshd/createClient.ts

```diff
@@ -40,41 +31,30 @@ import {
   resourceOneOfIsServer,
 } from 'teleterm/helpers';
 
-import { createFileTransferStream } from './createFileTransferStream';
-import { loggingInterceptor } from './interceptors';
+import { CloneableAbortSignal, cloneClient } from './cloneableClient';
 import * as types from './types';
 import {
-  ReportUsageEventRequest,
-  UnifiedResourceResponse,
   UpdateHeadlessAuthenticationStateParams,
+  UnifiedResourceResponse,
 } from './types';
-import createAbortController from './createAbortController';
-import { mapUsageEvent } from './mapUsageEvent';
+import { loggingInterceptor } from './interceptors';
 
 export function createTshdClient(
   addr: string,
   credentials: grpc.ChannelCredentials
 ): types.TshdClient {
   const logger = new Logger('tshd');
-  const tshd = new TerminalServiceClient(addr, credentials, {
+  const transport = new GrpcTransport({
+    host: addr,
+    channelCredentials: credentials,
     interceptors: [loggingInterceptor(logger)],
   });
+  const tshd = cloneClient(new TerminalServiceClient(transport));
```

Comment:
The next step would be to expose this through preload, right? I think the end goal would be to get rid of `createTshdClient` completely.

Reply:
> I think the end goal would be to get rid of createTshdClient completely.

Yes, that would be a next step. From now `createTshdClient` only converts one object to another and makes a request, we don't need this.